File: One Task/Task 1/c programming/52.java
import java.util.*; 

class FiftyTwo {
    public static void main(String args[]) {
        String days[] = { "SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT" }; 

        int date = 01;
        int month = 12;
        int year = 2023;

        Date d = new Date(year - 1901, month, date);

        Calendar c = Calendar.getInstance();
        c.setTime(d);

        for(int i=0; i<c.getActualMaximum(Calendar.DAY_OF_MONTH); i++) {
            int mon = (c.get(Calendar.MONTH) == 0)?12:c.get(Calendar.MONTH);
            System.out.println(days[c.get(Calendar.DAY_OF_WEEK) - 1]+" "+c.get(Calendar.DATE)+" "+mon+" "+c.get(Calendar.YEAR));
            c.add(Calendar.DATE, 1);
        }
    }
}

File: Without Node Module/Xampp/htdocs/core/week-1/c-programs/programs/52.php
<?php
    $month = 3;
    $year = 2024;

    echo "Calendar for $year - $month:<br>";
    echo date("F Y", strtotime("$year-$month-01")) . "<br>";

    $firstDay = date("N", strtotime("$year-$month-01"));
    $totalDays = cal_days_in_month(CAL_GREGORIAN, $month, $year);

    echo "<div class='d-flex justify-content-center'><table border='1'>";
    echo "<tr><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th><th>Sun</th></tr>";

    for ($i = 1 - $firstDay; $i <= $totalDays; $i++) {
        if ($i > 0) {
            echo "<td>$i</td>";
        } else {
            echo "<td></td>";
        }

        if ($i % 7 == 0) {
            echo "</tr><tr>";
        }
    }

    echo "</tr></table></div>";
?>
